Content.Server/Sound/EmitSoundSystem.cs

```diff
@@ -1,55 +1,50 @@
+using Content.Server.Interaction.Components;
+using Content.Server.Sound.Components;
+using Content.Server.Throwing;
 using Content.Shared.Audio;
 using Content.Shared.Interaction;
 using Content.Shared.Throwing;
-using Content.Server.Interaction.Components;
-using Content.Server.Throwing;
 using JetBrains.Annotations;
 using Robust.Shared.Audio;
 using Robust.Shared.GameObjects;
-using Robust.Shared.IoC;
-using Robust.Shared.Prototypes;
+using Robust.Shared.Log;
 using Robust.Shared.Player;
-using Robust.Shared.Random;
 
 namespace Content.Server.Sound
 {
+    /// <summary>
+    /// Will play a sound on various events if the affected entity has a component derived from BaseEmitSoundComponent
+    /// </summary>
     [UsedImplicitly]
     public class EmitSoundSystem : EntitySystem
     {
-        [Dependency] private readonly IPrototypeManager _prototypeManager = default!;
-        [Dependency] private readonly IRobustRandom _random = default!;
-
         /// <inheritdoc />
         public override void Initialize()
         {
             base.Initialize();
-            SubscribeLocalEvent<EmitSoundOnLandComponent, LandEvent>((eUI, comp, arg) => PlaySound(comp));
-            SubscribeLocalEvent<EmitSoundOnUseComponent, UseInHandEvent>((eUI, comp, arg) => PlaySound(comp));
-            SubscribeLocalEvent<EmitSoundOnThrowComponent, ThrownEvent>((eUI, comp, arg) => PlaySound(comp));
-            SubscribeLocalEvent<EmitSoundOnActivateComponent, ActivateInWorldEvent>((eUI, comp, args) => PlaySound(comp));
+            SubscribeLocalEvent<EmitSoundOnLandComponent, LandEvent>((eUI, comp, arg) => HandleEmitSoundOn(comp));
+            SubscribeLocalEvent<EmitSoundOnUseComponent, UseInHandEvent>((eUI, comp, arg) => HandleEmitSoundOn(comp));
+            SubscribeLocalEvent<EmitSoundOnThrowComponent, ThrownEvent>((eUI, comp, arg) => HandleEmitSoundOn(comp));
+            SubscribeLocalEvent<EmitSoundOnActivateComponent, ActivateInWorldEvent>((eUI, comp, args) => HandleEmitSoundOn(comp));
```

Comment:
This is a bad idea, just make your methods take in the three arguments, even if two of them go unused.